### PR TITLE
Add `safe rekey` command for creating new unseal keys in a Vault

### PR DESCRIPTION
**Pull request by geofffranks**
`safe rekey` now exists, to assist with re-keying a Vault with
new unseal keys. It prompts for the old unseal keys, and when
enough have been entered, the rekey occurs, and new unseal keys
are printed out to the user. `--num-unseal-keys` and `--keys-to-unseal`
can be used to configure how many unseal keys are created, and
how many are required to unseal the vault.

Additionally, GPG keys can be specified via the `--gpg` flag, to encrypt
the unseal keys. When using this mode, each unseal key is encrypted
by a different GPG key (you should specify more than one GPG key). Keys
are looked up from the local GPG keyring. Each encrypted unseal key is
output at the end, tied to the GPG key that was requested.

For example:

```
$ safe rekey --gpg user1@example.com --gpg user2@example.com
Your Vault has been re-keyed. Please take note of your new unseal keys and store them safely!
Unseal key for user1@example.com:
REDACTED
Unseal key for user2@example.com:
REDACTED
```

Signed-off-by: Steven Barry <sba30@allstate.com>

### CODE REVIEW

**Comment by thomasmmitchell** [comment]
It's taken me quite awhile to get what --num-unseal-keys and --keys-to-unseal actually correspond to (apparently `shares` and `threshold`, respectively, in Vault lingo). Can you add to the description in the help clarifying what each of these do? Also, there's no way to fetch what flags are actually required through the help at the moment, so could you expand the usage to actually show what a user should enter?

Also, can there be a confirmation message if a rekey is successfully cancelled?

And... does Vault actually restrict that you must have a threshold > 1 if shares > 1, or is that only imposed through safe?

**Comment by geofffranks** [comment]
Vault imposes the `threshold > 1` if `shares >1` with a weird message, so we had safe make it better.

I was trying for more human-understandable flags than Vault's lingo for `secret_shares` and `secret_threshold`. I'll add help for them, but if you have other ideas on makeing the flags more intuitive, I'm open to it

**Comment by thomasmmitchell** [comment]
While the names you've given are probably fine as long as you include something in the help about them, I'm tempted to suggest using something with the original vault terminology so that it's google-able if somebody is really lost on the concept of a rekey. That said... don't care much as long as its documented in the help.

In the case that a rekey is already started through some means when this is called, I believe /sys/rekey/init would error. Is there a specific error code we can catch and ask the user if they'd like to cancel the existing rekey or continue it? It's such an edge case that it's probably not needed, but if it'd be trivial, it'd add a little resiliency

**Comment by geofffranks** [comment]
We would have to inspect the text error messages returned via the API for specific strings, which apparently changed between 0.6.5 and 0.7.3, so I'm not sure we'd want to go through those hoops and have to keep updating the error message.

Maybe down the road, a `safe cancel rekey` command?

**Comment by thomasmmitchell** [comment]
Along with a `--continue` flag for `rekey`, I think that would be fine. That said, I don't think it's necessary for this PR, but I might put in an issue for it as low hanging fruit for someone down the road.

**Comment by geofffranks** [comment]
@thomasmmitchell Updated with the help messaging + cancellation confirmation you requested

File: main.go [hunk 1]
```diff
@@ -1284,8 +1284,27 @@ NBITS defaults to 2048.
 
 	r.Dispatch("rekey", &Help{
 		Summary: "Re-key your Vault with new unseal keys",
-		Usage:   "safe rekey [OPTIONS]",
+		Usage:   "safe rekey [--gpg email@address ...] [--num-unseal-keys <int>] [--keys-to-unseal <int>]",
 		Type:    DestructiveCommand,
+		Description: `
+Rekeys Vault with new unseal keys. This will require a quorum
+of existing unseal keys to accomplish. This command can be used
+to change the nubmer of unseal keys being generated via
```
Comment (thomasmmitchell) [inline main.go]: number* :)
Also, can we specify the defaults for not giving the numbers?

Comment (geofffranks) [reply main.go]: 👍

**Comment by geofffranks** [comment]
@thomasmmitchell amended with the typo fix + defaults

**Comment by geofffranks** [comment]
\o/ thanks!

**Comment by thomasmmitchell** [comment]
reference #103

**Pull request merged**